src/main/java/org/truffleruby/language/loader/RequireNode.java

```diff
@@ -130,10 +130,37 @@ protected boolean require(VirtualFrame frame, String feature,
                     featureLoader.ensureCExtImplementationLoaded(feature, callNode);
 
                     if (getContext().getOptions().CEXTS_LOG_LOAD) {
-                        Log.info("loading cext module %s", expandedPath);
+                        Log.info("loading cext module %s (requested as %s)", expandedPath, feature);
+                    }
+
+                    final CallTarget callTarget;
+
+                    try {
+                        callTarget = featureLoader.parseSource(source);
+                    } catch (JavaException e) {
+                        if (e.getCause() instanceof RuntimeException
+                                && e.getCause().getCause() instanceof IOException
+                                && e.getCause().getCause().getCause() instanceof UnsatisfiedLinkError) {
```

Comment:
Another way to make this cleaner would be to have a helper method that looks through causes in a loop and look for a UnsatisfiedLinkError.